### PR TITLE
feat: implement consumer with re-connect DEVOPS-130

### PR DESCRIPTION
**Pull request by teleivo**
* pinned im-manager and im-inspector to previous version of instance-queue; so we can update the consumer without affecting the im-manager.
* implemented consumer with re-connect
* readme shows you how to setup a local queue with a producer and consumer (using the instance-queue library)
* tested using integration tests with https://golang.testcontainers.org
* re-connect is integration tested using https://github.com/Shopify/toxiproxy ; which you can also play with in the local setup

Plan after merging this
* replace the existing consumer logic in the im-manager with the new consumer
* implement a producer reusing the re-connect logic
* replace the existing producer logic in the im-inspector with the new producer
* implement dead letter queue

### CODE REVIEW

**Comment by teleivo** [comment]
TestReconnectConsumerConnection seems to be flaky. Never failed locally. I thought it might be a candidate to become flaky when writing it. And 🎉 it is 😂 I will update it to be more robust

**Comment by teleivo** [comment]
> TestReconnectConsumerConnection seems to be flaky. Never failed locally. I thought it might be a candidate to become flaky when writing it. And tada it is joy I will update it to be more robust

Used Eventually to check that Consume() returns a temporary error while the underlying connection/channel are recovered. This should make the test robust. Lets see 😋

**Review by tonsV2 — APPROVED** [review]

**Review by radnov — APPROVED** [review]
Looks good to me (as far as I get it). Just added 2 minor questions.

File: cmd/consume/main.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+			break
+		}
+
+		var terr temporary
+		if !errors.As(err, &terr) {
+			return fmt.Errorf("error is not temporary: %w", err)
+		}
+
+		fmt.Printf("Consume failed: %s\n", err)
+		time.Sleep(time.Second * 5)
```
[the remark this answers is not in the document]
Comment (teleivo) [reply cmd/consume/main.go:59]: these are just example consumers and producers. thats why they are not configurable/tested

**Pull request merged**